meteor/server/api/showStyles.ts

```diff
@@ -67,63 +66,84 @@ export async function insertShowStyleBaseInner(organizationId: OrganizationId |
 		_rundownVersionHash: '',
 	}
 	ShowStyleBases.insert(showStyleBase)
-	await insertShowStyleVariantInner(showStyleBase, 'Default')
+
+	const showStyleVariant: ShowStyleVariant = {
+		_id: getRandomId(),
+		name: 'Default',
+		blueprintConfig: {},
+		_rundownVersionHash: '',
+		_rank: 500,
+		showStyleBaseId: showStyleBase._id,
+	}
+	await insertShowStyleVariant(showStyleVariant)
 	return showStyleBase._id
 }
-export async function insertShowStyleVariant(
-	context: MethodContext | Credentials,
-	showStyleBaseId: ShowStyleBaseId,
-	name?: string
-): Promise<ShowStyleVariantId> {
+async function assertShowStyleBaseAccess(context: MethodContext | Credentials, showStyleBaseId: ShowStyleBaseId) {
 	check(showStyleBaseId, String)
 
 	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
 	const showStyleBase = access.showStyleBase
 	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
+}
 
-	return insertShowStyleVariantInner(showStyleBase, name)
+export async function createDefaultShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleBaseId: ShowStyleBaseId,
+	rank?: number,
+	name?: string
+): Promise<ShowStyleVariantId> {
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
+
+	const showStyleVariant: ShowStyleVariant = {
+		_id: getRandomId(),
+		name: name || 'Variant',
+		blueprintConfig: {},
+		_rundownVersionHash: '',
+		_rank: rank || 500,
```

Comment:
You can use nullish coalescing `??` instead of the `||` operator, then you support the rank `0`.

Reply:
Changed to use the suggestion!